### ISSUE TITLE
Error when using workspace-defined package metadata.

### ISSUE THREAD

**Issue by m00nwtchr** [kellpossible/cargo-i18n#97]
IE: https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspacepackage-table
Specifically:
```
error: proc macro panicked
   --> XXXX/app/src/i18n.rs:31:9
    |
31  |         i18n_embed_fl::fl!($crate::i18n::LANGUAGE_LOADER, $message_id)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: XXXX/app/src/view/XXXXXX.rs:275:26
    |
275 |                     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
    |                          ------------------ in this macro invocation
    |
    = help: message: Error parsing Cargo.toml: InvalidManifest("`version` in [package] section is not a string")
    = note: this error originates in the macro `fll` (in Nightly builds, run with -Z macro-backtrace for more info
```
```toml
[package]
name = "XXXX"
version.workspace = true
authors.workspace = true
edition.workspace = true
license.workspace = true
```
Changing version to `version = "1.0.0"` fixes it.

**Comment by kellpossible** [comment]
Thanks for letting me know!

**Comment by kellpossible** [comment]
Personal note, I guess workspace support should also pay attention to the changes made in #93

**Comment by Umio-Yasuno** [comment]
Although it's not a complete fix, the patch below fixes the issue with building with fluent.  

```diff
diff --git a/i18n-embed-fl/src/lib.rs b/i18n-embed-fl/src/lib.rs
index 7bd8a31..b65ce7f 100644
--- a/i18n-embed-fl/src/lib.rs
+++ b/i18n-embed-fl/src/lib.rs
@@ -340,10 +340,15 @@ pub fn fl(input: TokenStream) -> TokenStream {
     let fluent_loader = input.fluent_loader;
     let message_id = input.message_id;
 
-    let manifest = find_crate::Manifest::new().expect("Error reading Cargo.toml");
-    let current_crate_package = manifest.crate_package().expect("Error parsing Cargo.toml");
-
-    let domain = current_crate_package.name;
+    let domain = {
+        let manifest = find_crate::Manifest::new().expect("Error reading Cargo.toml");
+        manifest.crate_package().map(|pkg| pkg.name).unwrap_or(
+            std::env::var_os("CARGO_PKG_NAME")
+                .expect("Error fetch `CARGO_PKG_NAME` env")
+                .to_string_lossy()
+                .to_string(),
+        )
+    };
 
     let domain_data = if let Some(domain_data) = DOMAINS.get(&domain) {
         domain_data
diff --git a/i18n-embed/i18n-embed-impl/src/lib.rs b/i18n-embed/i18n-embed-impl/src/lib.rs
index b8e556c..9b6a190 100644
--- a/i18n-embed/i18n-embed-impl/src/lib.rs
+++ b/i18n-embed/i18n-embed-impl/src/lib.rs
@@ -84,10 +84,15 @@ pub fn gettext_language_loader(_: proc_macro::TokenStream) -> proc_macro::TokenS
 #[cfg(feature = "fluent-system")]
 pub fn fluent_language_loader(_: proc_macro::TokenStream) -> proc_macro::TokenStream {
     let manifest = find_crate::Manifest::new().expect("Error reading Cargo.toml");
-    let current_crate_package = manifest.crate_package().expect("Error reading Cargo.toml");
+    let name = manifest.crate_package().map(|pkg| pkg.name).unwrap_or(
+        std::env::var_os("CARGO_PKG_NAME")
+            .expect("Error fetch `CARGO_PKG_NAME` env")
+            .to_string_lossy()
+            .to_string(),
+    );
 
     // Special case for when this macro is invoked in i18n-embed tests/docs
-    let i18n_embed_crate_name = if current_crate_package.name == "i18n_embed" {
+    let i18n_embed_crate_name = if name == "i18n_embed" {
         "i18n_embed".to_string()
     } else {
         manifest
@@ -131,7 +136,7 @@ pub fn fluent_language_loader(_: proc_macro::TokenStream) -> proc_macro::TokenSt
     let domain_str = config
         .fluent
         .and_then(|f| f.domain)
-        .unwrap_or(current_crate_package.name);
+        .unwrap_or(name);
     let domain = syn::LitStr::new(&domain_str, proc_macro2::Span::call_site());
 
     let gen = quote::quote! {
```

**Comment by Umio-Yasuno** [comment]
This issue still occurs because `i18n-embed` still depends on `i18n-embed-impl v0.8.1` and `i18n-embed-impl` crate has not been updated in `crates.io` .  

https://github.com/kellpossible/cargo-i18n/blob/master/i18n-embed/Cargo.toml#L26

**Comment by kellpossible** [comment]
Thanks for the heads up! @Umio-Yasuno I published a new version v0.8.4, that `i18n-embed` version `0.15.1` now depends on.

**Comment by Umio-Yasuno** [comment]
@kellpossible Thanks!
